pkg/mount/mount.go

```diff
@@ -78,17 +80,25 @@ func RecursiveUnmount(target string) error {
 		}
 		logrus.Debugf("Trying to unmount %s", m.Mountpoint)
 		err = Unmount(m.Mountpoint)
-		if err != nil && i == len(mounts)-1 {
-			if mounted, err := Mounted(m.Mountpoint); err != nil || mounted {
-				return err
+		if err != nil {
+			// If the error is EINVAL either this whole package is wrong (invalid flags passed to unmount(2)) or this is
+			// not a mountpoint (which is ok in this case).
+			// Meanwhile calling `Mounted()` is very expensive.
+			if err == syscall.EINVAL {
+				continue
+			}
+			if i == len(mounts)-1 {
```

Comment:
Is this a check for the top (i.e. final) mount? It seems it's wrong, as we are iterating through all mounts, not just the one provided by a user and its submounts.

In other words, I believe the correct check here has to be `m.Mountpoint == target`.

Reply:
Well, the path passed in may not even be a mount. The reason it's the final mount is the list is sorted in reverse-lexicographic order.